### ISSUE TITLE
Hieroglyphs in javadoc hover

### ISSUE THREAD

**Issue by spkoder** [redhat-developer/vscode-java#788]
Issue Type: <b>Bug</b>

Hello. I have a problem. I included another module as dependency to my maven java project. In this module I documented (javadoc) methods by russian symbols. Hints for this methods in this project are displayed as hieroglyphs. I think that extension decodes characters from sources of module using ANSI encoding (Win-1251), although the source itself is stored in UTF-8. Opening the same project in Idea, this problem is not observed. I tried to prescribe different settings, nothing helps.

Extension version: 0.38.0
VS Code version: Code 1.31.0 (7c66f58312b48ed8ca4e387ebd9ffe9605332caa, 2019-02-05T22:35:56.624Z)
OS version: Windows_NT x64 6.3.9600

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD FX(tm)-8350 Eight-Core Processor            (8 x 3991)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.96GB (2.98GB free)|
|Process Argv|C:\Users\hunter\Documents\!\Java\TestUseOtherModule|
|Screen Reader|no|
|VM|0%|

</details>

**Comment by spkoder** [comment]
[TestLogic.zip](https://github.com/redhat-developer/vscode-java/files/2842651/TestLogic.zip)
[TestUseOtherModule.zip](https://github.com/redhat-developer/vscode-java/files/2842652/TestUseOtherModule.zip)
Source of module:
![default](https://user-images.githubusercontent.com/15929530/52442832-c7c0e480-2b56-11e9-8b0f-3c307af45aae.png)
Trying to see the hint for method of module in other project:
![2](https://user-images.githubusercontent.com/15929530/52442833-c7c0e480-2b56-11e9-817e-eaf53099eeb7.png)
When I press F12 (Go to definition):
![3](https://user-images.githubusercontent.com/15929530/52442834-c7c0e480-2b56-11e9-9b7c-37b50fb7e2fc.png)
Then opens new tab (Logic.class):
![4](https://user-images.githubusercontent.com/15929530/52442838-c8597b00-2b56-11e9-808d-58a55a81a31f.png)

**Comment by snjeza** [comment]
Try to add 
```
"java.jdt.ls.vmargs": "-Dfile.encoding=UTF-8 -noverify -Xmx1G -XX:+UseG1GC -XX:+UseStringDeduplication"
```
to your VS Code settings.

**Comment by fbricon** [comment]
it's weird I see the same problem on my mac, which uses UTF-8 as system encoding by default. Explicitly adding the jvm flag fixes it (see #790)

**Comment by spkoder** [comment]
Thanks you. Flag -Dfile.encoding=UTF-8 helped me. 
Interestingly, setting this flag by default will not break existing projects?

**Comment by fbricon** [comment]
Fixed with #790. 

The file encoding used by the server is now vscode's own file encoding (see the `files.encoding` preference), which defaults to utf8.

Users can modify the value by changing the encoding in their settings.json. eg. : 
```
"{
	"files.encoding": "gbk",
}
```
which will be overridden by:
```
"[java]": {
	"files.encoding": "gbk",
}
```
which will be overridden by:
```
"{
	"jdt.ls.vmargs":"-Dfile.encoding=gbk ..."
}
```
